yarn-project/end-to-end/src/e2e_key_rotation.test.ts

```diff
@@ -0,0 +1,197 @@
+import { createAccounts } from '@aztec/accounts/testing';
+import {
+  type AztecAddress,
+  type AztecNode,
+  type DebugLogger,
+  ExtendedNote,
+  Fq,
+  Fr,
+  Note,
+  type PXE,
+  type Wallet,
+  computeSecretHash,
+  retryUntil,
+} from '@aztec/aztec.js';
+import { derivePublicKeyFromSecretKey } from '@aztec/circuits.js';
+import { KeyRegistryContract, TestContract, TokenContract } from '@aztec/noir-contracts.js';
+import { getCanonicalKeyRegistryAddress } from '@aztec/protocol-contracts/key-registry';
+
+import { jest } from '@jest/globals';
+
+import { expectsNumOfEncryptedLogsInTheLastBlockToBe, setup, setupPXEService } from './fixtures/utils.js';
+
+const TIMEOUT = 120_000;
+
+describe('e2e_key_rotation', () => {
+  jest.setTimeout(TIMEOUT);
+
+  let aztecNode: AztecNode | undefined;
+  let pxeA: PXE;
+  let pxeB: PXE;
+  let walletA: Wallet;
+  let walletB: Wallet;
+  let logger: DebugLogger;
+  let teardownA: () => Promise<void>;
+  let teardownB: () => Promise<void>;
+
+  let keyRegistryWithB: KeyRegistryContract;
+
+  let testContract: TestContract;
+
+  const SHARED_MUTABLE_DELAY = 5 + 3;
```

Comment:
Why such a long delay?

Reply:
Nuked this in https://github.com/AztecProtocol/aztec-packages/pull/6405/commits/968085851a7b24561726329b7310238d9e4c2fc9